### ISSUE TITLE
Load of helm chart catalog throws and exception if users cannot see helm repos

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#10156]
**Setup**
- Rancher version: v2.8.0

**Describe the bug**
- If a user doesn't have permission to view helm repos an error is thrown
- ```
  fetch.client.js:79 Error in fetch(): TypeError: Cannot read properties of undefined (reading 'filter')
    at _callee$ (catalog.js:357:1)
    at tryCatch (regeneratorRuntime.js:44:1)
    at Generator.<anonymous> (regeneratorRuntime.js:125:1)
    at Generator.next (regeneratorRuntime.js:69:1)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:22:1)
  ```
- Regression brought in by // https://github.com/rancher/dashboard/pull/8754/files#diff-31b0043fb6852fd3650396c7d7fbe3b5ae4ba1fe3a4293fadb00a712ac53684cR357

**To Reproduce**
- Create a user with only the global `user-base` role
- Assign them as a cluster owner to an rke1 cluster
- Log in as that user
- Go to Edit the rke1 cluster

**Result**
- Exception thrown

**Expected Result**
- No exception thrown

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem and/or errors in the browser's dev console -->

**Additional context**
- Given the user's roles the edit page won't display correctly, even after any fix is applied

**Comment by IsaSih** [comment]
@richard-cox It's a bit unclear what should be tested here. First, the issue describes that there's an error when a user with a user-base role tries to view helm repos. Then, the repro steps described the action of editing a cluster. With the user-base I created, I added it with the owner role of the local rke1 cluster. The user is then able not only to view the list of helm repos, but to edit the list. Also, is able to edit the cluster config with no errors. Is this the expected behavior?

**Comment by richard-cox** [comment]
The issue describes the underlying bug. There's ambiguity with `see` though, sorry. Here i just mean fetch rather than view in the UI. The create/edit cluster page will load helm charts via helm repos and if the user cannot fetch repos then the page would throw the error in the console. If there are no errors when editing the cluster with that user then everything is good

**Comment by IsaSih** [comment]
okay, so, there are no fetch error to the helm charts as seen in the following screenshot:

![Screenshot 2024-04-19 at 3.17.14 PM.png](https://zube.io/files/rancher/07bc4a35348bac4828f9f6b94f3d3f68-screenshot-2024-04-19-at-3-17-14-pm.png)

However, when the user-base user edits the cluster (I added an env variable: key foo / value bar ) , the following errors occur, though the UI saves the changes:
![Screenshot 2024-04-19 at 3.18.57 PM.png](https://zube.io/files/rancher/e336c965da811ab10f0f4c6ecc3a5d47-screenshot-2024-04-19-at-3-18-57-pm.png)![Screenshot 2024-04-19 at 3.20.16 PM.png](https://zube.io/files/rancher/549a09fde727758389457006a92c48f0-screenshot-2024-04-19-at-3-20-16-pm.png)

**Comment by richard-cox** [comment]
@IsaSih Those should fall under the bucket of general issues given the user's permission (sub-`standard user`). There's a few different issues floating around that will address them (starting with https://github.com/rancher/dashboard/issues/10787)

**Comment by IsaSih** [comment]
Tests pass

**Comment by rak-phillip** [comment]
This is the issue that fixes the problems detailed in #10866
